Run services as root user within docker container, and readme updates for windows development (#629)
centerofci/mathesar: README.md

If it's your first time running the application, you'll also need to run database migrations and install Mathesar types and functions:
```
docker exec mathesar_service_1 python manage.py migrate
docker exec -it mathesar_service_1 python install.py
```

You should now have a web server and database server running. Opening `http://localhost:8000` in your browser will open the application. For sample table data, you can create a new table in the UI using the `patents.csv` file found in `/mathesar/tests/data`. 

It is recommended that you keep the Docker containers running while you make changes to the code. Any change to the code made locally will sync to the container and the version deployed at `http://localhost:8000` will always be the latest local version of the code.

### Developing in Windows

> Windows users, who want to run Mathesar docker development environment in WSL, are adviced to clone the repository in a linux filesystem. WSL does not work well with HMR, required for frontend development, when the project resides in a windows filesystem. Please refer our [common issues page](https://wiki.mathesar.org/engineering/common-issues), and the [frontend development readme file](https://github.com/centerofci/mathesar/blob/master/mathesar_ui/README.md#developing-in-windows) for more details. 

Reviewer: ```suggestion
Windows users, who want to run Mathesar docker development environment in WSL, are adviced to clone the repository in a linux filesystem. WSL does not work well with HMR, required for frontend development, when the project resides in a windows filesystem. Please refer to our [common issues page](https://wiki.mathesar.org/engineering/common-issues), and the [frontend development readme file](https://github.com/centerofci/mathesar/blob/master/mathesar_ui/README.md#developing-in-windows) for more details. 
```

Reviewer 2: ```suggestion
Windows users who want to run the Mathesar Docker development environment in WSL are advised to clone the repository in a Linux filesystem. When the project resides in a Windows filesystem, WSL does not work well with hot module replacement (HMR), which is required for frontend development. Please refer to our [Common Issues wiki page](https://wiki.mathesar.org/engineering/common-issues), and the [frontend development README file](https://github.com/centerofci/mathesar/blob/master/mathesar_ui/README.md#developing-in-windows) for more details. 
```

Backend tests:
```
docker exec mathesar_service_1 pytest
```

Frontend tests:
```
docker exec mathesar_service_1 bash -c "cd mathesar_ui && npm test"
```

### Opening a shell in the container

If you need to do some work on the container that's running the code, here's how you access it:
```
docker exec -it mathesar_service_1 bash
```

> ### Common issues

Reviewer 2: ```suggestion
### Troubleshooting
```

> Please refer our [common issues wiki page](https://wiki.mathesar.org/engineering/common-issues) for common issues and steps to resolve them, while setting up and running Mathesar.

Reviewer 2: ```suggestion
Please refer to our [Common Issues wiki page](https://wiki.mathesar.org/engineering/common-issues) for instruction on troubleshooting common issues while setting up and running Mathesar.
```
